File: Easy/valid_parentheses.py
class Solution:
    def isValid(self, s: str) -> bool:
        # Open brackets position
        om = []

        # Close brackets mapping
        cm = {
            ')':'(',
            ']':'[',
            '}':'{',
        }

        for b in s:
            if b in '([{':
                om.append(b)
            else:
                if om and cm[b] == om[-1]:
                    om.pop(-1)
                else:
                    return False
            
        if not om:
            return True
        else:
            return False

File: Easy/Swift/valid_parentheses.swift
class Solution {
    func isValid(_ s: String) -> Bool {
        // Open brackets postition
        var om = [Character]()
        
        // Close brackets mapping
        let cm: [Character: Character] = [
            ")":"(",
            "]":"[",
            "}":"{"
        ]
        
        for b in s {
            if "([{".contains(b) {
                om.append(b)
            } else {
                if !om.isEmpty && cm[b] == om.last {
                    om.removeLast()
                } else {
                    return false
                }
            }
        }
        
        if !om.isEmpty { return false } else { return true }
    }
}
